regression-test/suites/nereids_rules_p0/infer_predicate/extend_infer_equal_predicate.groovy

```diff
@@ -377,4 +377,9 @@ suite("extend_infer_equal_predicate") {
     qt_pull_up_from_intersect """select a from(select a from (select t1.a from extend_infer_t3 t1 where t1.a<10 intersect select t2.a from extend_infer_t4 t2 where t2.a<10  ) tt
             limit 10) t  where a<10 order by 1 ;"""
     qt_pull_up_from_agg """select a from (select a from extend_infer_t3 t1 where a<10 group by a limit 10) t where a<10 order by 1"""
+
+    // test left join right table predicate pull up
+    qt_leftjoin_right_pull_up_shape """
+        explain shape plan select * from extend_infer_t3 t1 left join extend_infer_t4 t2 on t1.a=t2.a left join extend_infer_t5 t3 on t2.a= t3.a where t1.a=1;
```

Comment:
add more cases：
1. more t1 where condition: `t1.a is not null`, `t1.a in (1, 2)`
2. more table join:  t1 left join t2 left join t3 left join t4 ...  and condition is the middle table `where t2.a = 1`